src/codeflare_sdk/cluster/config.py

```diff
@@ -70,8 +113,60 @@ def __post_init__(self):
         self._memory_to_string()
         self._str_mem_no_unit_add_GB()
         self._memory_to_resource()
-        self._gpu_to_resource()
         self._cpu_to_resource()
+        self._gpu_to_resource()
+        self._combine_extended_resource_mapping()
+        self._validate_extended_resource_requests(self.head_extended_resource_requests)
+        self._validate_extended_resource_requests(
+            self.worker_extended_resource_requests
+        )
+
+    def _combine_extended_resource_mapping(self):
+        if overwritten := set(self.extended_resource_mapping.keys()).intersection(
+            DEFAULT_RESOURCE_MAPPING.keys()
+        ):
+            if self.overwrite_default_resource_mapping:
+                warnings.warn(
+                    f"Overwriting default resource mapping for {overwritten}",
+                    UserWarning,
+                )
+            else:
+                raise ValueError(
+                    f"Resource mapping already exists for {overwritten}, set overwrite_default_resource_mapping to True to overwrite"
+                )
+        self.extended_resource_mapping = {
+            **DEFAULT_RESOURCE_MAPPING,
+            **self.extended_resource_mapping,
+        }
+
+    def _validate_extended_resource_requests(
+        self, extended_resources: typing.Dict[str, int]
+    ):
+        for k in extended_resources.keys():
+            if k not in self.extended_resource_mapping.keys():
+                raise ValueError(
+                    f"extended resource '{k}' not found in extended_resource_mapping, available resources are {list(self.extended_resource_mapping.keys())}, to add more supported resources use extended_resource_mapping. i.e. extended_resource_mapping = {{'{k}': 'FOO_BAR'}}"
+                )
+
+    def _gpu_to_resource(self):
+        if self.head_gpus:
```

Comment:
There is a case here where I specified `num_gpus`, `head_gpus` and `head_extended_resource_requests={"amd.com/gpu": 1}`, `worker_extended_resource_requests={"amd.com/gpu": 2}`

The AMD GPU was overwritten by the `nvidia.com` GPU. Should we concern ourselves with this case or trust that the user will read the warning and remove head/num_gpus?

Reply:
It shouldn't overwrite, it would end up specifying both `nvidia.com/gpu=num_gpus` and `amd.com/gpu=2` in the extended resource dict. I was having head_gpus just add the nvidia_gpus in order to match previous functionality since we assumed it was nvidia by default before.

Reply:
I'm for appending the the accelerators with the default nvidia gpus for now 👍